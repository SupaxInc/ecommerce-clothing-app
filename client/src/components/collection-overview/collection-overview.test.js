import React from "react";
import { shallow } from "enzyme";

import * as redux from 'react-redux';
import CollectionOverview from "./collection-overview";
import CollectionPreview from "../collection-preview/collection-preview";

describe('CollectionOverview component', () => {
    // Arrange
    let wrapper;
    // Mocking collections
    let mockCollections = {
        1: { id: 1 }, 
        2: {id: 2}, 
        3: {id: 3}
    };
    // Mock shop state
    let mockState = {
        shop: {
            collections: mockCollections
        }
    }

    const useSelectorMock = jest.spyOn(redux, 'useSelector');

    beforeEach(() => {
        // Selector should change collections object to an array by mapping the mockCollections keys
        // Allows us to loop over the collections objects and create CollectionPreview items
        /*
            mockCollectionsArray = [{
                {id: 1}, {id: 2}, {id: 3}
            }]
        */
        useSelectorMock.mockImplementation((callback) => callback(mockState));

        wrapper = shallow(<CollectionOverview />);
    });

    it('should render CollectionOverview', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the same number of CollectionPreview as the collections array', () => {
        expect.assertions(1);
        expect(wrapper.find(CollectionPreview).length).toBe(3);
    });

})