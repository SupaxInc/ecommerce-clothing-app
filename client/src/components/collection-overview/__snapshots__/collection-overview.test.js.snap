// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionOverview component should render CollectionOverview 1`] = `ShallowWrapper {}`;
